### PR TITLE
AO3-6078 Unescape title HTML when notifying user about work deletion

### PR DESCRIPTION
**Pull request by JonathanMeans**
# Pull Request Checklist

* [X] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [X] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [X] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
* [X] Have you added the [Jira](https://otwarchive.atlassian.net) issue number
  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [X] Do you have fewer than 5 pull requests already open? If not, please wait
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6078 (please fill in issue number and remove this comment)

## Purpose

Show unescaped HTML for Title of fic in pop-up when deleted work

## Testing Instructions

How can the Archive's QA team verify that this is working as you intended?
Automated test was added

(If you have a Jira account with access, please update or comment on the issue
with any new or missing testing instructions.)

## References

Are there other relevant issues/pull requests/mailing list discussions?

## Credit

*What name do you want us to use to credit your work in the [Archive of Our Own's Release Notes](https://archiveofourown.org/admin_posts?tag=1)?*
Jonathan Means

(If you have a Jira account, please include the same name in the "Full name"
field on your Jira profile, so we can assign you the issues you're working on.)

*What pronouns do you prefer (she/her, he/him, zie/hir etc)?*
he/him

(If you do not fill in this section, we will use your GitHub account name and
they/them pronouns to avoid making assumptions about your gender.)

---

The HTML characters are still escaped in the email it sends. Not sure y'all's preference for that behavior, or if you want it included in this work if it is a bug.

Don't have permissions to assign myself stuff in JIRA, so just opening a PR and hopefully no one wastes too much time.

### CODE REVIEW

**Comment by sarken** [comment]
Hi, Jonathan!

Thank you so much for this pull request. Someone will be along to review it soon.

In the meantime, I've set the Jira issue to "In Review," so no one should mistakenly create a duplicate pull request. If you'd like to create a Jira account, we can set you up with permissions for commenting on and assigning issues -- you can just reply here with the account name. (It makes things a bit easier for us on the organizational side if the Full Name on your Jira account either closely matches your GitHub name or includes it in parentheses, e.g. "Nickname (Jonathan Means).")

>The HTML characters are still escaped in the email it sends. Not sure y'all's preference for that behavior, or if you want it included in this work if it is a bug.

Thanks for the heads-up! That sounds like a bug to me. We usually try to keep pull requests as small as possible (though we don't always succeed 😆), so no need to add the fix to this one, but I will definitely make an issue so we don't forget about it.

**Comment by JonathanMeans** [comment]
Thanks! I've just signed up for Jira: https://otwarchive.atlassian.net/jira/people/5ff5a999849d640111493936

Could I get permissions to add labels or whatever it is I need to do on GitHub as well, please?

**Comment by sarken** [comment]
Great! I've added you to the contributors group to give you access and made you the issue's assignee as long as I was there.

Unfortunately, GitHub labels are set up so they can only be added by users with write access to the repo, so someone with that access will drop by and add it to any new PRs as soon as they have a second.

Regarding the escaped HTML in the emails, could you clarify which email you're seeing that in and where (e.g. body, subject)? I did a quick test of the deleted work email and it looks like the HTML is properly unescaped in the body. (The attachment names are another story, but we're going to completely rework the attachments in the future.)

**Comment by JonathanMeans** [comment]
Oh, my bad I was just looking at line 21 of the test:

> And "newbie" should be notified by email about the deletion of "All Hell &lt;b&gt;Breaks&lt;/b&gt; Loose"

But I guess it ends up with the right characters. I haven't figured out setting up a test account in the console quite yet, so was just relying on the automated tests, sorry!

File: app/controllers/works_controller.rb [hunk 1]
```diff
@@ -451,8 +452,9 @@ def destroy
     begin
       was_draft = !@work.posted?
       title = @work.title
+      unescaped_title = CGI.unescapeHTML(title)
       @work.destroy
-      flash[:notice] = ts('Your work %{title} was deleted.', title: title)
+      flash[:notice] = ts("Your work %{title} was deleted.", title: unescaped_title)
```
Comment (sarken) [inline app/controllers/works_controller.rb:457]: The good news is this is totally the right way to do this!

However, we're pretty deep into doing in the wrong way and just calling `.html_safe` on the `flash[:notice]`. After some brief discussion, we decided we'd prefer to be consistently wrong and use `.html_safe` until we can change the underlying issue with how we're saving these special characters in the database.

**Review by sarken — CHANGES_REQUESTED** [review]
Thanks for bearing with us on this! And no worries on the email -- it's always better to hear about an apparent bug than not, even if everything turns out to be fine.

**Comment by JonathanMeans** [comment]
IIUC, the security warning is unrelated to my code. Looks like there's already an issue open for it here: https://otwarchive.atlassian.net/browse/AO3-6107

**Review by sarken — APPROVED** [review]
Thank you!

**Pull request merged**